client/src/auth.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useState,
+} from "react";
+import { AUTH_STORAGE_KEY } from "@/utils/constants";
+import { UserPublic } from "@/types/types";
+export interface AuthContext {
+  isAuthenticated: boolean;
+  logout: () => Promise<void>;
+  user: UserPublic | null;
+  setSession: (user: UserPublic | null) => void;
+}
+
+const AuthContext = createContext<AuthContext | null>(null);
+
+const saveToStorage = (session: UserPublic | null) => {
+  if (session) {
+    localStorage.setItem(AUTH_STORAGE_KEY, JSON.stringify(session));
+  } else {
+    localStorage.removeItem(AUTH_STORAGE_KEY);
+  }
+};
+
+const loadFromStorage = (): UserPublic | null => {
+  // if (import.meta.env.DEV)
```

Comment:
should this remain commented for later or should we do something about this before merging?